compose/ui/ui-graphics/src/skikoMain/kotlin/androidx/compose/ui/graphics/layer/SkiaGraphicsLayer.skiko.kt

```diff
@@ -265,6 +293,7 @@ actual class GraphicsLayer internal constructor() {
 
             if (clip || shadowElevation > 0f) {
                 canvas.save()
+                restoreCount++
 
                 when (val outline = internalOutline) {
                     is Outline.Rectangle ->
```

Comment:
Instead of casting we can use the same approach as in `RenderNodeLayer`:

```
    private fun Canvas.clipRoundRect(rect: RoundRect, clipOp: ClipOp = ClipOp.Intersect) {
        val antiAlias = true
        nativeCanvas.clipRRect(rect.toSkiaRRect(), clipOp.toSkia(), antiAlias)
    }


    private fun ClipOp.toSkia() = when (this) {
        ClipOp.Difference -> ClipMode.DIFFERENCE
        ClipOp.Intersect -> ClipMode.INTERSECT
        else -> ClipMode.INTERSECT
    }
```

Reply:
Let's limit the scope of this fixing particular bug PR

Reply:
Never mind, I see that method was added exactly for this call earlier.